lms/djangoapps/course_api/blocks/views.py

```diff
@@ -337,6 +339,65 @@ def list(self, request, hide_access_denials=False):  # pylint: disable=arguments
         return response
 
 
+@method_decorator(transaction.non_atomic_requests, name='dispatch')
+@view_auth_classes(is_authenticated=False)
+class BlockMetadataView(DeveloperErrorViewMixin, ListAPIView):
+    """
+    **Use Case**
+
+        Returns the block metadata. Large data like index_dictionary related to
+        a block should be fetched using this API.
+
+    **Example requests**:
+
+        GET /api/courses/v1/block_metadata/<usage_id>/?
+            &include=index_dictionary
+
+    **Parameters**:
+
+        * "include": a comma-separated list of keys to include.
+          Valid keys are "index_dictionary".
+
+    **Response Values**
+
+        A dictionary of block_id, block_type and keys in "include" param.
+
+        * index_dictionary: (dict) The index_dictionary JSON data
+          (usually this is the text content of the block, for search
+          indexing or other purposes) for this block. Returned only if
+          the "index_dictionary" is included in the "include"
+          parameter.
+    """
+
+    def list(self, request, usage_key_string):  # pylint: disable=arguments-differ
+        """
+        Retrieves the usage_key for the requested block, and then returns the
+        block metadata
+
+        Arguments:
+            request - Django request object
+        """
+
+        try:
+            usage_key = UsageKey.from_string(usage_key_string)
+        except InvalidKeyError:
+            raise ValidationError(f"'{str(usage_key_string)}' is not a valid usage key.")  # lint-amnesty, pylint: disable=raise-missing-from
+
+        if not has_access(request.user, "staff", usage_key):
```

Comment:
I think this could also be just user access, not staff access. But it's fine to leave as staff-only for now.

Reply:
Had to select `staff` role as that is the only option available for `UsageKey`.
https://github.com/openedx/edx-platform/blob/dccb792f739032183e15c870fabd061ec45c9f78/lms/djangoapps/courseware/access.py#L603-L618